torch/jit/annotations.py

```diff
@@ -281,10 +279,7 @@ def try_ann_to_type(ann, loc):
     if inspect.isclass(ann):
         if hasattr(ann, "__torch_script_class__"):
             return ClassType(_qualified_name(ann))
-        # Why Callable?  forward is declared to be a Callable so that
-        # people can define it without mypy complaining.  But we shouldn't
-        # try to recursively compile it!
-        ignored_builtin_classes = (torch.nn.Module, tuple, list, Callable)
```

Comment:
@ezyang do you remember what caused you to add `Callable` ? removing it doesn't seem to break any tests.

Reply:
Not sure if it helps, but from the PR description from https://github.com/pytorch/pytorch/pull/38211

> The Callable trick on forward caused TorchScript to stop performing inference on the forward body, which is bad because in general we can only figure out the most accurate type by doing TorchScript inference. I added a special case to infer_type to ensure we always do inference for Module.forward, even if it is annotated (which it is), and another special case to make sure we ignore references to Callable (which we shouldn't process) recursively.